src/Optimisers.jl

```diff
@@ -63,8 +63,33 @@ or [`update!`](@ref).
 
 # Example
 ```jldoctest
-julia> Optimisers.setup(Descent(0.1f0), (x = rand(3), y = (true, false), z = tanh))
-(x = Leaf(Descent{Float32}(0.1), nothing), y = (nothing, nothing), z = nothing)
+julia> m = (x = rand(3), y = (true, false), z = tanh);
+
+julia> Optimisers.setup(Momentum(), m)  # same field names as m
+(x = Leaf(Momentum{Float32}(0.01, 0.9), [0.0, 0.0, 0.0]), y = (nothing, nothing), z = nothing)
+```
+
+The recursion into structures uses Functors.jl, and any new `struct`s containing parameters
+need to be marked with `Functors.@functor` before use. Further refinements can, if necessary,
```

Comment:
```suggestion
need to be marked with `Functors.@functor` before use.
See [the Flux docs](https://fluxml.ai/Flux.jl/stable/models/advanced/#Customising-Parameter-Collection-for-a-Model) for more guidance on how to do this for layers specifically.
Further refinements can, if necessary,
```
I was going to link to `@functor` too, but looks like we need a new Flux docs build for that to render.